homeassistant/components/incharge/sensor.py

```diff
@@ -0,0 +1,152 @@
+"""Get status for all stations for InCharge."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+import datetime
+import logging
+from typing import Any
+
+from incharge.api import InCharge
+import requests
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME, UnitOfEnergy
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import UpdateFailed
+from homeassistant.util import Throttle
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = datetime.timedelta(minutes=5)
+
+
+@dataclass
+class InChargeSensorEntityDescription(SensorEntityDescription):
+    """Describes InCharge sensor entity."""
+
+    precision: int | None = None
```

Comment:
Why not use: suggested_display_precision

https://developers.home-assistant.io/blog/2023/02/08/sensor_presentation_rounding